examples/07_join_aggregation.py

```diff
@@ -0,0 +1,387 @@
+"""
+Self-aggregation with MovieLens
+===============================
+
+MovieLens is a famous movie dataset used for both explicit
+and implicit recommender systems. It provides a main table,
+"ratings", that can be viewed as logs or transactions, comprised
+of only 4 columns: ``userId``, ``movieId``, ``rating`` and ``timestamp``.
+MovieLens also gives a contextual table "movies", including
+``movieId``, ``title`` and ``types``, to enable content-based feature extraction.
+
+In this notebook, we only deal with the main table "ratings".
+Our objective is **not to achieve state-of-the-art performance** on
+the explicit regression task, but rather to illustrate how to perform
+feature engineering in a simple way using |AggJoiner| and |AggTarget|.
+Note that our performance is higher than the baseline of using the mean
+rating per movies.
+
+
+.. |AggJoiner| replace::
+     :class:`~skrub.AggJoiner`
+
+.. |AggTarget| replace::
+     :class:`~skrub.AggTarget`
+
+.. |TableVectorizer| replace::
+     :class:`~skrub.TableVectorizer`
+
+.. |DatetimeEncoder| replace::
+     :class:`~skrub.DatetimeEncoder`
+
+.. |TargetEncoder| replace::
+     :class:`~skrub.TargetEncoder`
+
+.. |make_pipeline| replace::
+     :class:`~sklearn.pipeline.make_pipeline`
+
+.. |Pipeline| replace::
+     :class:`~sklearn.pipeline.Pipeline`
+
+.. |GridSearchCV| replace::
+     :class:`~sklearn.model_selection.GridSearchCV`
+
+.. |TimeSeriesSplit| replace::
+     :class:`~sklearn.model_selection.TimeSeriesSplit`
+
+.. |HGBR| replace::
+     :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
+"""
+
+###############################################################################
+# The data
+# --------
+#
+# We begin with loading the ratings table from MovieLens.
+# Note that we use the light version (100k rows).
+import pandas as pd
+import numpy as np
+
+from skrub.datasets import fetch_movielens
+
+
+ratings = fetch_movielens(dataset_id="ratings")
+ratings = ratings.X.sort_values("timestamp").reset_index(drop=True)
+ratings["timestamp"] = pd.to_datetime(ratings["timestamp"], unit="s")
+
+X = ratings[["userId", "movieId", "timestamp"]]
+y = ratings["rating"]
+print(X.shape)
+print(X.head())
+
+###############################################################################
+# Encoding the timestamp with a TableVectorizer
+# ---------------------------------------------
+#
+# Our first step is to extract features from the timestamp, using the
+# |TableVectorizer|. Natively, it uses the |DatetimeEncoder| on datetime
+# columns, and doesn't interact with numerical columns.
+from skrub import TableVectorizer, DatetimeEncoder
+
+
+table_vectorizer = TableVectorizer(
+    datetime_transformer=DatetimeEncoder(add_day_of_the_week=True)
+)
+table_vectorizer.set_output(transform="pandas")
+X_date_encoded = table_vectorizer.fit_transform(X)
+print(X_date_encoded)
```

Comment:
I used `print` because I had issues with column display. Columns were overlapping on my local rendering of the example. The online doc doesn't suffer from this though. I will follow your recommendation.